events/janus_gelfevh.c

```diff
@@ -0,0 +1,476 @@
+
+		.incoming_event = janus_gelfevh_incoming_event,
+		.handle_request = janus_gelfevh_handle_request,
+
+		.events_mask = JANUS_EVENT_TYPE_NONE
+	);
+
+/* Plugin creator */
+janus_eventhandler *create(void) {
+	JANUS_LOG(LOG_VERB, "%s created!\n", JANUS_GELFEVH_NAME);
+	return &janus_gelfevh;
+}
+
+
+/* Useful stuff */
+static volatile gint initialized = 0, stopping = 0;
+static GThread *handler_thread;
+static void *janus_gelfevh_handler(void *data);
+static janus_mutex evh_mutex;
+
+/* JSON serialization options */
+static size_t json_format = JSON_INDENT(3) | JSON_PRESERVE_ORDER;
+
+/* Queue of events to handle */
+static GAsyncQueue *events = NULL;
+static json_t exit_event;
+static void janus_gelfevh_event_free(json_t *event) {
+	if(!event || event == &exit_event)
+		return;
+	json_decref(event);
+}
+
+/* Gelf backend to send the events to */
+static char *backend = NULL;
+static char *port = NULL;
+static int sockfd;
+//static struct sockaddr_in servaddr;
+
+/* Parameter validation (for tweaking via Admin API) */
+static struct janus_json_parameter request_parameters[] = {
+	{"request", JSON_STRING, JANUS_JSON_PARAM_REQUIRED}
+};
+static struct janus_json_parameter tweak_parameters[] = {
+	{"events", JSON_STRING, 0},
+	{"backend", JSON_STRING, 0},
+	{"port", JSON_STRING, 0}
+};
+/* Error codes (for the tweaking via Admin API */
+#define JANUS_GELFEVH_ERROR_INVALID_REQUEST		411
+#define JANUS_GELFEVH_ERROR_MISSING_ELEMENT 	412
+#define JANUS_GELFEVH_ERROR_INVALID_ELEMENT 	413
+#define JANUS_GELFEVH_ERROR_UNKNOWN_ERROR 		499
+
+/* Plugin implementation */
+static void janus_gelfevh_connect(void) {
+	struct addrinfo *res = NULL;
+	janus_network_address addr;
+	janus_network_address_string_buffer addr_buf;
+	struct sockaddr_in servaddr;
+
+	if(getaddrinfo(backend, NULL, NULL, &res) != 0 ||
+		janus_network_address_from_sockaddr(res->ai_addr, &addr) != 0 ||
+		janus_network_address_to_string_buffer(&addr, &addr_buf) != 0) {
+		if(res)
+			freeaddrinfo(res);
+		JANUS_LOG(LOG_ERR, "Could not resolve address (%s)...\n", backend);
+		return;
+	}
+	const char *host = g_strdup(janus_network_address_string_from_buffer(&addr_buf));
+	freeaddrinfo(res);
+
+    if((sockfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0 ) {
+		JANUS_LOG(LOG_ERR, "Socket creation failed");
+		return;
+	}
+
+	memset(&servaddr, 0, sizeof(servaddr));
+
+	servaddr.sin_family = AF_INET;
+	servaddr.sin_port = htons(atoi(port));
+	servaddr.sin_addr.s_addr = inet_addr(host);
+
+	if(connect(sockfd, (struct sockaddr *)&servaddr, sizeof(servaddr)) < 0) {
+		JANUS_LOG(LOG_WARN, "Connect to Gelf host failed\n");
+		return;
+	}
+	JANUS_LOG(LOG_INFO, "Connected to GELF backend: [%s:%s]\n", host, port);
+}
+
+static int janus_gelfevh_send(char *message) {
+	//JANUS_LOG(LOG_WARN, "Sending event to GELF: %s\n", message);
+	if(!message) {
+		JANUS_LOG(LOG_WARN, "Message is NULL, not sending to Gelf!\n");
+		return -1;
+	}
+	if (write(sockfd, message, MAX_GELF_LOG_LEN) < 0) {
```

Comment:
Code style: please remove space between `if` and bracket.

Reply:
Why are you trying to send `MAX_GELF_LOG_LEN` bytes, when the message is very likely shorter?

Reply:
Also, not sure why you're using `write` instead of `send`, which you can use now that the socket is connected. Reading around, it looks like `send` gives more info when it fails (which we do care about, see above on socket errors).